### ISSUE TITLE
[feature request] auto disassembly of components past quota

### ISSUE THREAD

**Issue by Kodi4444** [Gorea235/TaledensInvManagerUpdated#24]
Is it possible to add deconstructing of excess components? I know its possible because I found a script that is similar to TIM and can deconstruct items past quota (Isy's Inventory Manager https://steamcommunity.com/sharedfiles/filedetails/?id=1216126863) but it's not as good as TIM. and it would be way better if I could use only TIM for all my assembly, disassembly and sorting. instead of using TIM and Isy's Inventory Manager together.
Please.

**Comment by Der-Isy** [comment]
Not as good -__- I'm shocked! But thanks for the link ;)

**Comment by LordMike** [comment]
@Der-Isy could you put Isys inventory manager up on Github?

**Comment by smooreace** [comment]
TIM never had a shot against Isys work! That's fairly obvious too considering the conversation!

**Comment by Der-Isy** [comment]
@LordMike No, sorry.. I don't want any uncontrolled spinoffs of my script..

@smooreace Thanks *blushed* ;-)

**Comment by luvies** [comment]
I can look into it, but I plan on rewriting this script first. I'll make sure I'll keep it in mind though, since it seems like a useful opt-in system.

**Comment by LordMike** [comment]
@Der-Isy too bad.. It'd be great to have some features in Isys that are missing, such as more fine-grained control of resources - such as placing Stone specifically in ejectors, and not any ore type.

**Comment by smooreace** [comment]
Simple solution.... use the script that actually works! I used TIM for a
long time. Someone told me I was missing out.... they were most definitely
correct!

**Comment by Akuukis** [comment]
TIM and IIM are essentially the same things.. My 50 cents is that it would make sense that everyone could collaborate together instead of re-inventing the wheel.

@Der-Isy, I'd challange your argument of not having spin-offs, because not allowing to collaborate on IIM will make copy-cats instead (e.g. TIM), which are both less controllable for you and worse for users than spin-offs.

**Comment by smooreace** [comment]
Thats some pretty weak psychology there! Failed spinoffs do nothing but
draw more attention to the product/s that work!

When Kia came to be, did we see the world line up to buy the next POS out
of Korea? Or did we see a stronger returning customer base? Hint: Kia
floundered miserably until they made a product worthy of attention.

Collaboration is exactly what feeds the copy-cat market! The more people
that know the secrets, the more people there are that have the ability to
run off and do the same thing on their own.

**Comment by luvies** [comment]
Collaboration is the whole point of OSS, and it's why most things are moving towards it. Being open and allowing people to help if they want makes products better. Granted, you can't stop people from taking the code, but unless they make something better (which is good in-and-of itself), then it really doesn't detract from the initial product. There's a reason things like .NET Core are entirely open source (under MIT no less). And claiming that there are 'secrets that need to be kept' is really 'better-than-thou' thinking. Even something as important as OpenSSL, which power a huge proportion of the HTTPS-based internet, is completely open source.

That's not to say open source 'is best', I just prefer it. Hence why all my stuff (at least all the more recent stuff) is under MIT. Some people would prefer to keep their work more secure, which is fine (although it's something that I personally don't get behind).

Either way, it seems this issue has derailed a bit, so I'm going to lock it. I'll likely revisit this issue once I've worked out how to progress with TIM as a whole, though.
